artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/RemotingConnectionImpl.java

```diff
@@ -387,4 +388,15 @@ public void setClientID(String cID) {
    public String getClientID() {
       return clientID;
    }
+
+   @Override
+   public void killMessage(SimpleString nodeID) throws Exception {
+      if (clientVersion < DisconnectConsumerWithKillMessage.VERSION_INTRODUCED) {
+         throw new Exception(String.valueOf(clientVersion));
```

Comment:
Can this be a more specific ActiveMQ Exception?  Even if its just `ActiveMQException`